tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/DiagnosticsTest.java

```diff
@@ -79,4 +97,37 @@ private void assertDiagnostic(Diagnostic diagnostic, Object[] expected) {
         assertEquals(diagnostic.location().lineRange().startLine().line(), expected[1]);
         assertEquals(diagnostic.location().lineRange().startLine().offset(), expected[2]);
     }
+
+    private Object[][] getExpectedDiagnosticProperties() {
+        return new Object[][] {
+                {"[string]|record {| string arg2; |}", UNION},
+                {"()", NIL},
+        };
+    }
+
+    private void assertDiagnosticProperties(List<Diagnostic> diagnostics) {
+        int index = 0;
+
+        for (Diagnostic diagnostic : diagnostics) {
+
+            List<DiagnosticProperty<?>> diagnosticProperties = diagnostic.properties();
+
+            if (diagnosticProperties != null) {
```

Comment:
Let's use early exits. So we can minimize the nested indentations.

Reply:
Updated.